controllers/operands/networkAddons_test.go

```diff
@@ -456,52 +456,58 @@ var _ = Describe("CNA Operand", func() {
 		})
 
 		type ovsAnnotationParams struct {
-			annotationExists  bool
+			ovsExists         bool
+			setAnnotation     bool
 			annotationValue   string
 			ovsDeployExpected bool
 		}
 		DescribeTable("when reconciling ovs-cni", func(o ovsAnnotationParams) {
-			hcoOVSConfig := commonTestUtils.NewHco()
-			hcoOVSConfig.Annotations = map[string]string{}
-
-			if o.annotationExists {
-				hcoOVSConfig.Annotations["deployOVS"] = o.annotationValue
+			existingCNAO, err := NewNetworkAddons(hco)
+			Expect(err).ToNot(HaveOccurred())
+			if o.ovsExists {
+				existingCNAO.Spec.Ovs = &networkaddonsshared.Ovs{}
 			}
 
-			existingResource, err := NewNetworkAddons(hcoOVSConfig)
-			Expect(err).ToNot(HaveOccurred())
+			if o.setAnnotation {
+				hco.Annotations = map[string]string{
+					"deployOVS": o.annotationValue,
+				}
+			}
 
-			cl := commonTestUtils.InitClient([]runtime.Object{hco, existingResource})
+			cl := commonTestUtils.InitClient([]runtime.Object{hco, existingCNAO})
 			handler := (*genericOperand)(newCnaHandler(cl, commonTestUtils.GetScheme()))
 			res := handler.ensure(req)
 			Expect(res.UpgradeDone).To(BeFalse())
 			Expect(res.Err).ToNot(HaveOccurred())
 
-			foundResource := &networkaddonsv1.NetworkAddonsConfig{}
+			foundCNAO := &networkaddonsv1.NetworkAddonsConfig{}
 			Expect(
 				cl.Get(context.TODO(),
-					types.NamespacedName{Name: existingResource.Name, Namespace: existingResource.Namespace},
-					foundResource),
-			).ToNot(HaveOccurred())
+					types.NamespacedName{Name: existingCNAO.Name, Namespace: existingCNAO.Namespace},
+					foundCNAO),
+			).To(Succeed())
 
 			if o.ovsDeployExpected {
-				Expect(existingResource.Spec.Ovs).ToNot(BeNil(), "Ovs spec should be added")
+				Expect(foundCNAO.Spec.Ovs).ToNot(BeNil(), "OVS spec should be added")
 			} else {
-				Expect(existingResource.Spec.Ovs).To(BeNil(), "Ovs spec should not be added")
+				Expect(foundCNAO.Spec.Ovs).To(BeNil(), "OVS spec should not be added")
 			}
 		},
-			Entry("should have ovs if deployOVS annotation is set to true", ovsAnnotationParams{
-				annotationExists:  true,
+			Entry("should have OVS if deployOVS annotation is set to true", ovsAnnotationParams{
+				ovsExists:         false,
+				setAnnotation:     true,
 				annotationValue:   "true",
 				ovsDeployExpected: true,
 			}),
 			Entry("should not have ovs if deployOVS annotation is not set to true", ovsAnnotationParams{
```

Comment:
Please consider "is not set to true" -> "is set to false"

Reply:
not necessarily. OVS is deployed if the annotation value is exactly `"true"`, otherwise it is considered as false and OVS won't be deployed.

Reply:
right but this case does set, and it set to false, so better be explicit

Reply:
Fixed. Added two new cases:
"should not have ovs if deployOVS annotation is not set to true" with some other value, and
"should not have ovs if deployOVS annotation is empty" with an empty string.